bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/security/TokenKey.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mideaac.internal.security;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link TokenKey} returns the active Token and Key.
+ *
+ * @author Jacek Dobrowolski - Initial Contribution
+ * @author Bob Eckhoff - JavaDoc
+ */
+@NonNullByDefault
+public class TokenKey {
```

Comment:
This calss can be turned into a record:
```
public record TokenKey (String token, String key) {}
```

Reply:
Yes, this is fixed. It can be improved further. The body now has two methods `getToken()` and `getKey()` they can be removed. The record has default getters to its members. For example whenever you use `record.getKey()` somwhere in the code you can change this to `record.key()`

Reply:
The record idea is new to me (as are most things ;-)).  If I remove the getters I have a problem with this test. Do records get imported differently?
```
    public void testTokenKey() {
        config.token = "D24046B597DB9C8A7CA029660BC606F3FD7EBF12693E73B2EF1FFE4C3B7CA00C824E408C9F3CE972CC0D3F8250AD79D0E67B101B47AC2DD84B396E52EA05193F";
        config.key = "97c65a4eed4f49fda06a1a51d5cbd61d2c9b81d103ca4ca689f352a07a16fae6";
        TokenKey tokenKey = new TokenKey(config.token, config.key);
        String tokenTest = tokenKey.getToken();
        String keyTest = tokenKey.getKey();
        assertEquals(config.token, tokenTest);
        assertEquals(config.key, keyTest);
```
with this import;
`import org.openhab.binding.mideaac.internal.security.TokenKey;`

Reply:
Out of my head, it looks like 
```
String tokenTest = tokenKey.getToken();
String keyTest = tokenKey.getKey();
```
should become
```
String tokenTest = tokenKey.token();
String keyTest = tokenKey.key();
```

Reply:
awesome.  That will fixed in next commit.